diff --git a/core/modules/update/src/Form/UpdateManagerInstall.php b/core/modules/update/src/Form/UpdateManagerInstall.php
index b7afcbad76..4fb8a18398 100644
--- a/core/modules/update/src/Form/UpdateManagerInstall.php
+++ b/core/modules/update/src/Form/UpdateManagerInstall.php
@@ -245,7 +245,8 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     // install the code, there's no need to prompt for FTP/SSH credentials.
     // Instead, we instantiate a Drupal\Core\FileTransfer\Local and invoke
     // update_authorize_run_install() directly.
-    if (fileowner($project_real_location) == fileowner($this->sitePath) && !$test_authorize) {
+    // if (fileowner($project_real_location) == fileowner($this->sitePath) && !$test_authorize) {
+    if (!$test_authorize) {
       $this->moduleHandler->loadInclude('update', 'inc', 'update.authorize');
       $filetransfer = new Local($this->root, \Drupal::service('file_system'));
       $response = call_user_func_array('update_authorize_run_install', array_merge([$filetransfer], $arguments));
